geode-core/src/main/java/org/apache/geode/cache/query/internal/CompiledLike.java

```diff
@@ -349,6 +349,30 @@ private String getRegexPattern(String pattern) {
             sb.append(ch);
           }
           break;
+        case '$': // replace with [$]
+          if (prevMetaChar) {
+            sb.append('\\');
+            sb.append('E');
+            prevMetaChar = false;
+          }
+
+          numConsecutiveBackSlash = 0;
+          for (int j = i - 1; j > -1; --j) {
+            if (pattern.charAt(j) == '\\') {
+              ++numConsecutiveBackSlash;
+            } else {
+              break;
+            }
+          }
+
+          // Check if the $ has a valid escape. Backtrack to check for \.
```

Comment:
Why "$" needs to be escaped being a literal in like predicate value. The special characters that are used in LIKE or "_", "%" and "\" these chars needs to be escaped with escape char "\". 
I think we should be able to compare the "$" literal within the like predicate without escaping it. Please let me know if I am missing anything.

Reply:
When it use the compare operation it checks string, but when using LIKE operation it check character by character, and when comes to "$" character it present it like endline character and cannot got right results. Now with escaping it will check if string has "$" character and compare it like any other character.